### ISSUE TITLE
Evaluation results are different from the paper

### ISSUE THREAD

**Issue by chingswy** [DarlingHang/st-nerf#7]
Hello authors, thanks for your pretrained model. I render the images of walking and ｔａｅｋｗｏｎｄｏ sequences and calculate the PSNR and SSIM metrics.

||PSNR|SSIM|
|----|----|----|
|walking | 22.8 | 0.91 |
|taekwondo | 29.3 | 0.95 | 

They are slightly different from your paper. Is that normal?

**Comment by Jiakai-Zhang** [comment]
It should not have such big differences for walking sequence. We will check our rendered result and the quantitative result soon, thanks for your question!

**Comment by chingswy** [comment]
Can you also share your evaluation code if possible?

**Comment by DarlingHang** [comment]
For quantitative results in our paper, we calculated the average PSNR in the 16 capture views for each timestamp. We do have evaluation code, however, it is not yet supported by this repository and we will do it as soon as possible. 

As a reminder, if you are using our demo scripts to do evaluation, you need to modify some lines of codes to make sure you are using correct camera pose and timestamp.

Can you describe how you are calculating the metrics so we can help you?

**Comment by shenwenhao01** [comment]
I have opened a pull request where we do evaluation using the two modified demos and the evaluation part is in "demo/st_eval.py". By the way, I improved the results on walking demo to '23.4' in PSNR metric and '0.92' in SSIM metric by deleting the line "neural_renderer.set_near(0.1)". So could you help us check out the evaluation code and provide some clues if possible?

**Comment by DarlingHang** [comment]
I have checked out your code. Here are some suggestions:
1. For the walking sequence, you need to remove the codes for inverting camera poses. （line 53 of walking_demo.py）
2. In line 51 of walking_demo.py, try modifying `neural_renderer.set_pose_duration(i,15) ` into `neural_renderer.set_pose_duration(i,i+1) `. It is similar for taekwondo.
3. You can check your rendered images before computing metrics. Also, you need to make sure that the cam pose is the same as the one for ground truth within each iteration.

**Comment by shenwenhao01** [comment]
I followed your suggestions and rendered walking demo again. But unfortunately, the result was almost the same. 
Actually, I find that the rendering result's time sequence is not synchronized with provided ground truth, which makes me quite confused.
For example, the first image below is our rendering result for cam7/frame45, and the second one is the corresponding ground truth.
![45](https://user-images.githubusercontent.com/56423956/140450237-2f26e954-a17a-419d-a914-57332fe374f5.jpg)
![000045](https://user-images.githubusercontent.com/56423956/140450241-3a070132-0cc4-486c-8c59-e8777f0b235a.png)

**Comment by DarlingHang** [comment]
I checked the config file, in the `config_walking.yml`, the `FRAME_NUM` is set to 50, which means only first 50 frames should be used for training and testing.  You may want to try modifying line 52 of `walking_demo.py` to `neural_renderer.set_smooth_path_poses(50, around=False)`.

I am not 100 percent sure this will give the same results as ours. Please feel free to talk with us if there are further issues.

**Comment by Jiakai-Zhang** [comment]
> I checked the config file, in the `config_walking.yml`, the `FRAME_NUM` is set to 50, which means only first 50 frames should be used for training and testing. You may want to try modifying line 52 of `walking_demo.py` to `neural_renderer.set_smooth_path_poses(50, around=False)`.
> 
> I am not 100 percent sure this will give the same results as ours. Please feel free to talk with us if there are further issues.

There is also a `FRAME_OFFSET: 25` in config file, so when we change the line 52 of demo file, we may expect the **first rendered image** is corresponding to the **26th gt image**, the reason we skipped these frames is that the performers are almost static.

**Comment by shenwenhao01** [comment]
I followed your advice and only render 50 frames of the walking sequence. Then I compared the latest rendered images with those before and I see no obvious differences by my eyes.
When I evaluate the results quantitively, I also considered the `FRAME_OFFSET=25` and only uses 50 frames each view. However, the results should drops to `psnr: 23.092209980440142, ssim: 0.9132714219777885`! I guess that it is the move of two people that makes the rendering images less sharp. 
For example, the image below is our rendering result for cam15/frame49 and the corresponding ground truth.
![49](https://user-images.githubusercontent.com/56423956/140602530-d9a4c9df-9274-4db6-a176-85c54167c53e.jpg)
![000074](https://user-images.githubusercontent.com/56423956/140602532-85040b52-3c80-48d2-8e61-4c5288a85626.png)

**Comment by Jiakai-Zhang** [comment]
> I followed your advice and only render 50 frames of the walking sequence. Then I compared the latest rendered images with those before and I see no obvious differences by my eyes.
> When I evaluate the results quantitively, I also considered the `FRAME_OFFSET=25` and only uses 50 frames each view. However, the results should drops to `psnr: 23.092209980440142, ssim: 0.9132714219777885`! I guess that it is the move of two people that makes the rendering images less sharp.
> For example, the image below is our rendering result for cam15/frame49 and the corresponding ground truth.
> ![49](https://user-images.githubusercontent.com/56423956/140602530-d9a4c9df-9274-4db6-a176-85c54167c53e.jpg)
> ![000074](https://user-images.githubusercontent.com/56423956/140602532-85040b52-3c80-48d2-8e61-4c5288a85626.png)

It’s a little weird that the psnr is just improved a little comparing with the first time you rendered viewpoint and timestamp is all wrong for corresponding gt image, and even dropped comparing with the second time the result timestamp is wrong. I just see no reason it should happen. But I think your rendered result may be right for now, could you please check is there something wrong when you evaluate the metrics?

**Comment by shenwenhao01** [comment]
OK, I'll check closely. Thank you!

**Comment by Jiakai-Zhang** [comment]
There is no further question so I closed this issue. Feel free to re-open it.

**Comment by sycamore27** [comment]
> Hello authors, thanks for your pretrained model. I render the images of walking and ｔａｅｋｗｏｎｄｏ sequences and calculate the PSNR and SSIM metrics.
> 
> PSNR	SSIM
> walking	22.8	0.91
> taekwondo	29.3	0.95
> They are slightly different from your paper. Is that normal?

Hello, can you share the demo dataset of walking and taekwondo? The link in GitHub is empty. Thank you very much!!!
